src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1839,6 +1826,9 @@ MultiExtendedOpNode(Query *queryTree)
 	extendedOpNode->hasDistinctOn = queryTree->hasDistinctOn;
 	extendedOpNode->hasWindowFuncs = queryTree->hasWindowFuncs;
 	extendedOpNode->windowClause = queryTree->windowClause;
+	extendedOpNode->onlyPushableWindowFunctions =
+		!queryTree->hasWindowFuncs ||
+		SafeToPushdownWindowFunction(queryTree, NULL);
```

Comment:
For the following query
```
SELECT
  user_id,
  sum(jsonb_array_length(payload->'commits')) AS commits,
  RANK () OVER (
    PARTITION BY user_id
    ORDER BY
      sum(jsonb_array_length(payload->'commits')) DESC
  )
FROM github_events
GROUP BY user_id
HAVING
  sum(jsonb_array_length(payload->'commits')) > 0
ORDER BY commits DESC
LIMIT 10;
```

on the schema/data described on our docs: https://docs.citusdata.com/en/v9.2/get_started/tutorial_realtime_analytics.html

I do get a very strange pointer to `queryTree` here, which causes the `SafeToPushdownWindowFunction` to not return true.

Tracing how the code came here I have the following backtrace:
```
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 156.1
  * frame #0: 0x000000010bccbf0d citus.so`SafeToPushdownWindowFunction(query=0x00007fc1c2874270, errorDetail=0x0000000000000000) at query_pushdown_planning.c:400:34
    frame #1: 0x000000010bcc07a1 citus.so`MultiExtendedOpNode(queryTree=0x00007fc1c2874270) at multi_logical_planner.c:1831:3
    frame #2: 0x000000010bccc472 citus.so`SubqueryPushdownMultiNodeTree(queryTree=0x00007fc1c2874270) at query_pushdown_planning.c:1623:36
    frame #3: 0x000000010bccc1a1 citus.so`SubqueryMultiNodeTree(originalQuery=0x00007fc1c2874270, queryTree=0x00007fc1c2874388, plannerRestrictionContext=0x00007fc1c287e878) at query_pushdown_planning.c:522:20
    frame #4: 0x000000010bcbf59a citus.so`MultiLogicalPlanCreate(originalQuery=0x00007fc1c2874270, queryTree=0x00007fc1c2874388, plannerRestrictionContext=0x00007fc1c287e878) at multi_logical_planner.c:152:20
    frame #5: 0x000000010bcb5992 citus.so`CreateDistributedPlan(planId=34, originalQuery=0x00007fc1c2874270, query=0x00007fc1c2874388, boundParams=0x0000000000000000, hasUnresolvedParams=<unavailable>, plannerRestrictionContext=0x00007fc1c287e878) at distributed_planner.c:1068:31
    frame #6: 0x000000010bcb562a citus.so`CreateDistributedPlannedStmt(planContext=0x00007ffee6ab3f70) at distributed_planner.c:671:3
    frame #7: 0x000000010bcb465e citus.so`PlanDistributedStmt(planContext=0x00007ffee6ab3f70, rangeTableList=0x00007fc1c118f168, rteIdCounter=<unavailable>) at distributed_planner.c:602:24
    frame #8: 0x000000010bcb4168 citus.so`distributed_planner(parse=0x00007fc1c2874388, cursorOptions=<unavailable>, boundParams=<unavailable>) at distributed_planner.c:258:14
    frame #9: 0x000000010942dea8 postgres`pg_plan_query(querytree=0x00007fc1c2874388, cursorOptions=256, boundParams=0x0000000000000000) at postgres.c:878:9
    frame #10: 0x000000010942dfbe postgres`pg_plan_queries(querytrees=<unavailable>, cursorOptions=256, boundParams=0x0000000000000000) at postgres.c:968:11
    frame #11: 0x000000010942fd0e postgres`exec_simple_query(query_string="SELECT\n  user_id,\n  sum(jsonb_array_length(payload->'commits')) AS commits,\n  RANK () OVER (\n    PARTITION BY user_id\n    ORDER BY\n      sum(jsonb_array_length(payload->'commits')) DESC\n  )\nFROM github_events\nGROUP BY user_id\nHAVING \n  sum(jsonb_array_length(payload->'commits')) > 0\nORDER BY commits DESC\nLIMIT 10;") at postgres.c:1143:19
    frame #12: 0x000000010942f594 postgres`PostgresMain(argc=<unavailable>, argv=<unavailable>, dbname=<unavailable>, username=<unavailable>) at postgres.c:0
    frame #13: 0x00000001093ba1b6 postgres`BackendRun(port=0x000000010978b61c) at postmaster.c:4437:2
    frame #14: 0x00000001093b9a7f postgres`BackendStartup(port=<unavailable>) at postmaster.c:4128:3
    frame #15: 0x00000001093b9235 postgres`ServerLoop at postmaster.c:1704:7
    frame #16: 0x00000001093b72cf postgres`PostmasterMain(argc=3, argv=0x00007fc1bfc02280) at postmaster.c:1377:11
    frame #17: 0x00000001093213c6 postgres`main(argc=3, argv=0x00007fc1bfc02280) at main.c:228:3
    frame #18: 0x00007fff654237fd libdyld.dylib`start + 1
```

you see the query tree being passed around from `SubqueryMultiNodeTree(originalQuery=0x00007fc1c2874270, ...)` where the argument is `originalQuery`. However the actual original query resolves the partition column to `varattno: 7` (`userid`) where the query passed in here resolves it to `varattno: 1` (`event_id`). Causing the SafeToPushdown to return false here.

Besides causing a fault in not pushing down the window function I am more worried about it inadvertly resolving these varattno's where the grouping is on column 7 but the actual distribution is on column 1. It would cause a pushdown where the pushdown is not possible.

I think we by accident either do not pass the actual original query around (the pointer for the original query is different during my debugging session than the one passed in here), or we modify the original query. Eitherway, what we believe to be the original query here is a modified tree that resolves the varattno's incorrectly.